platform/x11/os_x11.cpp

```diff
@@ -1273,7 +1276,10 @@ Size2 OS_X11::get_screen_size(int p_screen) const {
 	XineramaScreenInfo *xsi = XineramaQueryScreens(x11_display, &count);
 
 	// Check if screen is valid
-	ERR_FAIL_INDEX_V(p_screen, count, Size2i(0, 0));
+	if (p_screen < 0 || p_screen >= count) {
+		XFree(xsi);
+		ERR_FAIL_V_MSG(Point2i(0, 0), vformat("Index %d is out of bounds (count = %d)", p_screen, count));
```

Comment:
Should be `Size2i` (same as `Point2i` but for consistency with the previous code). It's weird that these return int variants while the signature uses float ones though.